server/src/test/Parsing.test.ts

```diff
@@ -41,6 +41,35 @@ describe('Parsing Tests', () => {
             );
             done();
         });
+        it('should parse without errors an assignment with the `switchExprExclusive` instruction', (done) => {
+            checkInputValidityForRule(
+                (parser) => parser.instruction_assignment(),
+                `
+myVal = switchExprExclusive {
+    case(label == "a") : 1
+    case(label == "ab") : -1
+    case(label == "abc") : -1
+    default : 0
+}
+`,
+            );
+            done();
+        });
+        it('should parse without errors an assignment with the `switchExpr` instruction', (done) => {
+            checkInputValidityForRule(
+                (parser) => parser.instruction_assignment(),
+                `
+finalVal = switchExpr( myValue ) {
+    case CASE_1 : val1
+    case CASE_2 : val2
+    case CASE_3 : val3
+    case CASE_4 : val4
+};
+`,
```

Comment:
there is a confusion here:
`xxxExclusive` only means that if more than one case can be activated an error is produced. It can be used in both switch with or without variable.
All in all, you can have:
`switchExpr`, `switchExpr(val)`, `switchExprExclusive`, `switchExprExclusive(val)`.

Of course, the use of `xxxExclusive` is more relevant with a `switch`  that doesn't have a variable since you could have overlapping conditions easily if you don't pay attention

Reply:
Nice! It will make the grammar simpler :smile:

Reply:
@davanessoff-yseop I rewrote the test case, should be better now. Well, it should fail for any number greater than 0.